service-base/src/test/java/org/eclipse/hono/service/http/DefaultFailureHandlerTest.java

```diff
@@ -137,4 +144,34 @@ public void testHandlerWritesDetailMessageAndErrorCodeToBody() {
         verify(response).end();
     }
 
+    /**
+     * Verifies that the handler writes the status code and the status message to the response in case of failure and
+     * ctx.failure() returns null.
+     */
+    @Test
+    public void testHandlerOnDefaultWritesStatusCodeAsMessageToBody() {
+        final RoutingContext ctx = mock(RoutingContext.class);
+        final HttpServerRequest request = mock(HttpServerRequest.class, Mockito.RETURNS_MOCKS);
+        final HttpServerResponse response = mock(HttpServerResponseImpl.class, RETURNS_MOCKS);
```

Comment:
we usually do not use partial mocks, so please use the interface type instead

Reply:
This particular issue with status code = 401 and status message ="OK" is closely related to the implementation of HttpServerResponseImpl in vertx. In order to test that, I used partial mocks involving those methods.